src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -411,22 +412,19 @@ public List<int> PartialOrFullValidationFailed(ChainedHeader chainedHeader)
         }
 
         /// <inheritdoc />
-        public List<int> ConsensusTipChanged(ChainedHeader newConsensusTip)
+        public List<int> ConsensusTipChanged(ChainedHeader newConsensusTip, bool blockMined = false)
```

Comment:
Hmm one of the things I was trying to avoid is to pass this flag to the tree, I think we can do it in another way, perhaps a flag on the ChainedHeader that says this is a mined block?